TEE suppression plugin RPC copies full request buffer when actual output is smaller, leaking secure heap data

Finder16 opened the pull request:
fix bug

jforissier commented:
Thanks for the patch. Please provide a proper commit subject and description.

jforissier commented:
> Copies only the smaller of the plugin’s reported output length (outlen) and the requested length into the core/user buffers, preventing leftover heap data from leaking when the response is shorter than len

Excellent. I'd like to see this in the git commit description.

jforissier commented:
> Is this the correct way to do it?

Please run:
```
git config user.name "Jaeyeong Lee"
git config user.email "nakamurajames123@gmail.com"
git commit --amend -s --reset-author
```
This will set the Git `Author` field properly and in addition it will append a `Signed-off-by:` tag.

Finder16 committed:
fix: prevent leaking stale heap data when plugin output is shorter than requested

Copies only the smaller of the plugin’s reported output length (outlen)
and the requested length. This prevents leaking leftover secure-world
heap data when the plugin returns fewer bytes than the caller requested.

Previously, tee_invoke_supp_plugin_rpc() copied `len` bytes regardless of
the actual number of bytes produced by the plugin, resulting in kernel /
secure heap disclosure into the REE.

Signed-off-by: Jaeyeong Lee <nakamurajames123@gmail.com>

jenswi-linaro commented:
This is a bug worth fixing. Can we get back on track, please?

jenswikl commented:
@Finder16 has stopped responding. Could someone please create a new PR with this patch addressing my review comments?

Emantor commented:
Yes, see #7815.
